doc/sphinx-guides/source/installation/config.rst

```diff
@@ -1050,6 +1050,33 @@ Before being moved there,
   on your machine, large file uploads via API will cause RAM and/or swap usage bursts. You might want to point this to
   a different location, restrict maximum size of it, and monitor for stale uploads.
 
+.. _cache-rate-limiting:
+
+Configure Your Dataverse Installation to use JCache (with Hazelcast as provided by Payara) for Rate Limiting
+------------------------------------------------------------------------------------------------------------
+
+Rate limiting has been added to prevent users from over taxing the system either deliberately or by runaway automated processes.
+Rate limiting can be configured on a tier level with tier 0 being reserved for guest users and tiers 1-any for authenticated users.
+Superuser accounts are exempt from rate limiting.
+Rate limits can be imposed on command APIs by configuring the tier, the command, and the hourly limit in the database.
+Two database settings configure the rate limiting.
+Note: If either of these settings exist in the database rate limiting will be enabled. If neither setting exists rate limiting is disabled.
+
+- RateLimitingDefaultCapacityTiers is the number of calls allowed per hour if the specific command is not configured. The values represent the number of calls per hour per user for tiers 0,1,...
+  A value of -1 can be used to signify no rate limit. Also, by default, a tier not defined would receive a default of no limit.
```

Comment:
Currently there is no API to change the rate limit tier for a user. Guests will be 0 and auth users will be 1. We would need to add more logic to allow the admins to change the tier of a user. I agree it can get very complicated which is why it shouldn't be done unless absolutely necessary. I envision setting a user to tier 2 to give them no limit. 
 @scolapasta  @landreev  please give your opinions. This kind of logic would require another issue/task IMO

Reply:
Sorry, missed this question earlier. 
@poikilotherm Could you please clarify what kind of human error you had in mind? I'm assuming your comment was in response to "_Also, by default, a tier not defined would receive a default of no limit_." But per @stevenwinship reply, "_admin sets a non-existing rate tier for a user_" doesn't apply, as we don't even have a functionality yet to assign users to specific tiers. 
So, is the "human error" in question that the admin meant to set some default for, say, authenticated users, but failed to do so? Or if they misspelled the name of a command they wanted to limit in the next setting - `:RateLimitingCapacityByTierAndAction`?
I don't believe the settings api tries to validate any values of any other settings, so I agree that we probably don't want to try and do that for this one. 
@stevenwinship what happens if I set a limit for a tier number that does not exist in `:RateLimitingCapacityByTierAndAction`?  - Will there be a warning in the logs "No such tier" from the service bean code that uses the setting? What about a command that does not exist? - A warning in the logs like that would be the extent of any attempts to detect human errors from an admin within the scope of this PR, IMO. 

Apologies in case I'm confused/missing something myself.

Reply:
There is also a possibility of "_a tier not defined would receive a default of no limit_" simply being slightly confusing, and maybe it can be phrased in some more precise way. 
_If only one number is specified, it will be configured as the limit for tier 0 (guest users), with the other tiers left unlimited by default. I.e., `-d "1000"` is equivalent to `-d "1000,-1,-1"` etc. ..._ - idk, something like that. 

Same disclaimer as the above - sorry if I'm not reading it right myself. In which case, disregard.

Reply:
_"what happens if I set a limit for a tier number that does not exist in :RateLimitingCapacityByTierAndAction?"_
I'm a bit confused by this question. RateLimitingCapacityByTierAndAction is where you set the limit for the tier per command. If you set it then it does exist. Maybe I can explain the flow a little better:

A user has a tier (guest=0 auth=1-> infinity.
The RateLimitingCapacityByTierAndAction setting is looked at first to see if the tier and command exist. If yes then that rate is used. If No, then the RateLimitingDefaultCapacityTiers setting is checked (100,200,etc). So tier 0 would default to 100 call/hour, tier 1 defaults to 200 and anything above tier 1 would default to No Limit. If it's not defined in that setting then it would default to No Limit. This should only happen if rate limiting is turned off (no settings at all) or if you decide to give a user tier 3 which has to rate limit defined in the settings. 

As far as typos. I would think this would be the responsibility of the person setting up rates. There are no warnings. There is no list of commands to validate against and no code to verify. This makes adding new commands and removing commands easy to handle since it's ok to have commands that aren't used yet or that are no longer used.
Defining a tier/command setting doesn't need any user to be assigned the tier or the command to be valid. If command validation is needed or wanted then we can add that later. There are other questions that need to be answered such as what commands does the system need to run regardless of rate limiting?

Reply:
_If only one number is specified, it will be configured as the limit for tier 0 (guest users), with the other tiers left unlimited by default. I.e., -d "1000" is equivalent to -d "1000,-1,-1" etc. ... - idk, something like that._

I'll update the doc

Reply:
Saw the commits - that's perfect.

Reply:
(My question above about human errors was discussed and answered during standup yesterday. Specifying some random tier number, as in `{"tier": 13, "limitPerHour": 10, "actions": ["GetLatestPublishedDatasetVersionCommand"]}` in `:RateLimitingCapacityByTierAndAction` doesn't appear to be an issue. And if a user misspells the name of a command - `GteLatstUpblishedDatasetVersionCommand` - nothing happens, it'll just be ignored. I feel like it would be fairly easy to detect that and print a warning in the logs (not at the time of creating the setting, but later in the service bean that uses it). But since we generally make it the responsibility of the admin to not misspell things in virtually all other config situations, I don't feel like insisting that this needs to be done in the scope of this PR.)